crates/ide/src/ide/assists/convert_to_inherit.rs

```diff
@@ -14,9 +14,28 @@ pub(super) fn convert_to_inherit(ctx: &mut AssistsCtx<'_>) -> Option<()> {
         return None;
     }
 
-    // RHS should be a single identifier.
-    let rhs = match binding.value()?.flatten_paren()? {
-        ast::Expr::Ref(rhs) => rhs,
+    // RHS should be either:
+    // - A single identifier.
+    // - Or a select expression ending with a single (static) identifier.
+    let (from_expr, rhs) = match binding.value()?.flatten_paren()? {
+        ast::Expr::Ref(rhs) => ("".into(), rhs.token()?.text().into()),
+        ast::Expr::Select(rhs) if rhs.or_token().is_none() => {
+            let mut attrs = rhs.attrpath()?.attrs().collect::<Vec<_>>();
+            let mut from_expr = format!(" ({}", rhs.set()?.syntax());
+
+            let rhs = match AttrKind::of(attrs.pop()?) {
+                AttrKind::Static(Some(rhs)) => rhs,
+                _ => return None,
+            };
+
+            for attr in attrs {
+                from_expr.push('.');
+                from_expr.push_str(&attr.syntax().to_string());
+            }
+            from_expr.push(')');
```

Comment:
The from-expr seems to be `&src[rhs.syntax().text().start()..attrs.last()?.syntax().text().end()]`, where `src` is `ctx.db.file_content(file)`. So we can simplify this recollection.